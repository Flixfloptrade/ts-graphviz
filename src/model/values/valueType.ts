// tslint:disable:max-classes-per-file
// tslint:disable: prettier

export abstract class GraphVizValue {
  constructor(protected value: any) {}
  public toDot(): string {
    return this.value.toString();
  }
}

export class ArrowTypeValue extends GraphVizValue {
}
const a = new ArrowTypeValue({});
export class AspectTypeValue extends GraphVizValue {
}
export class BoolValue extends GraphVizValue {
}
export class ClusterModeValue extends GraphVizValue {
}
export class ColorValue extends GraphVizValue {
}
export class DirTypeValue extends GraphVizValue {
}
export class DoubleValue extends GraphVizValue {
}
export class EscStringValue extends GraphVizValue {
}
export class IntValue extends GraphVizValue {
}
export class LayerListValue extends GraphVizValue {
}
export class LayerRangeValue extends GraphVizValue {
}
export class LblStringValue extends GraphVizValue {
}
export class OutputModeValue extends GraphVizValue {
}
export class PackModeValue extends GraphVizValue {
}
export class PagedirValue extends GraphVizValue {
}
export class PointValue extends GraphVizValue {
}
export class PointfValue extends GraphVizValue {
}
export class PointfListValue extends GraphVizValue {
}
export class PortPosValue extends GraphVizValue {
}
export class QuadTypeValue extends GraphVizValue {
}
export class RankdirValue extends GraphVizValue {
}
export class RankTypeValue extends GraphVizValue {
}
export class RectValue extends GraphVizValue {
}
export class ShapeValue extends GraphVizValue {
}
export class SmoothTypeValue extends GraphVizValue {
}
export class StartTypeValue extends GraphVizValue {
}
export class StringValue extends GraphVizValue {
}
export class StyleValue extends GraphVizValue {
}
export class ViewPortValue extends GraphVizValue {
}
