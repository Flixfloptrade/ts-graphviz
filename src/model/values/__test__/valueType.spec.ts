import { LblStringValue } from '../valueType';
describe('valueType', () => {
  describe('LblStringValue', () => {
    describe('plain string', () => {
      test.each([['test']])('toDot result should be quoted', input => {
        const value = new LblStringValue(input);
        const dot = value.toDot();
        expect(dot).toMatch(/".+"/);
        expect(dot).not.toMatch(/<.+>/);
      });
    });

    describe('HTML Like string', () => {
      test.each([
        ['<<B>hoge</B>>'],
        [
          `<
          <U>hoge</U>
          >`,
        ],
        [
          `
          <<I>hoge</I>
          >
          `,
        ],
      ])('toDot result should not be quoted', input => {
        const value = new LblStringValue(input);
        const dot = value.toDot();
        expect(dot).not.toMatch(/".+"/);
        expect(dot).toMatch(/<.+>/);
      });
    });
  });
});
