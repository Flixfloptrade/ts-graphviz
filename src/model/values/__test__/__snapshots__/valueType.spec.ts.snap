// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valueType LblStringValue plain string toDot result should be quoted 1`] = `"\\"test\\""`;

exports[`valueType LblStringValue plain string toDot result should be quoted 2`] = `"\\"\\\\n          test\\\\n          \\""`;
